### PR TITLE
configure MaxRequestBodySize via environment variable

### LINKED ISSUE #7204: The request size limits need to be configurable
We need to use an environment variable allow users to override the request size limit. The environment variable name should be `FUNCTIONS_REQUEST_BODY_SIZE_LIMIT`

https://github.com/Azure/azure-functions-host/blob/69111926ee920d4ba10829c8fa34303bb8165a42/src/WebJobs.Script.WebHost/Program.cs#L43-L50

### PR DESCRIPTION
**Pull request by soninaren**
### Issue describing the changes in this PR

resolves #7204

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [x] Otherwise: Documentation issue linked to PR: MicrosoftDocs/azure-docs#72449
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        private long _maxRequestBodySize;
+
+        public HttpRequestBodySizeMiddleware(RequestDelegate next, IEnvironment environment)
+        {
+            _next = next;
+            _invoke = (context) =>
+            {
+                string bodySizeLimit = environment.GetEnvironmentVariable(FunctionsRequestBodySizeLimit);
+
+                if (!environment.IsPlaceholderModeEnabled() && long.TryParse(bodySizeLimit, out _maxRequestBodySize))
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs]: Will sync on improvements offline.

**Review by fabiocav — CHANGES_REQUESTED** [review]

File: test/WebJobs.Script.Tests/Middleware/HttpRequestBodySizeMiddlewareTests.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            httpContext.Features.Set<IHttpMaxRequestBodySizeFeature>(new TestHttpMaxRequestBodySizeFeature());
+            await middleware.Invoke(httpContext);
+            Assert.True(nextInvoked);
+
+            if (inPlaceHolderMode)
+            {
+                Assert.Equal(middleware.InnerInvoke, middleware.InvokeBeforeSpecialization);
+            }
+
+            if (!inPlaceHolderMode && isLimitSet)
```
Comment (sidkri) [inline test/WebJobs.Script.Tests/Middleware/HttpRequestBodySizeMiddlewareTests.cs:128]: Maybe also verify the value of "configuredLimit" here and also when !isLimitSet?  It's redundant with the test MaxRequestBodySizeLimit_RequestBodySizeLimit_SetExpectedValue but a little more thorough.

**Review by sidkri — DISMISSED** [review]

**Review by sidkri — DISMISSED** [review]

File: src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        internal RequestDelegate InnerInvoke { get => _invoke; }
+
+        public Task Invoke(HttpContext context)
+        {
+            return _invoke(context);
+        }
+
+        internal Task InvokeAfterSpecialization(HttpContext httpContext)
+        {
+            httpContext.Features.Get<IHttpMaxRequestBodySizeFeature>().MaxRequestBodySize = _maxRequestBodySize;
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs]: Unlikely to hit in our known scenarios, but a good practice is not to assume the feature is present, retrieve and check for `null`.  If it is, the option for this middleware is either to throw with a clear exception (making diagnostic easy) or noop. I would vote for the former.

File: src/WebJobs.Script/ScriptConstants.cs [hunk 4]
```diff
@@ -184,6 +184,7 @@ public static class ScriptConstants
         public static readonly ImmutableArray<string> AssemblyFileTypes = ImmutableArray.Create(".dll", ".exe");
         public static readonly string HostUserAgent = $"azure-functions-host/{ScriptHost.Version}";
         public static readonly NuGetVersion ExtensionBundleVersionTwo = new NuGetVersion("2.0.0");
+        public static readonly long? DefaultMaxRequestBodySize = 104857600;
```
Comment (fabiocav) [inline src/WebJobs.Script/ScriptConstants.cs]: Can you clarify again what API required you to define this as nullable?

**Comment by fabiocav** [comment]
Tests still seem to be failing on this

**Comment by sidkri** [comment]
Could you link to the documentation PR before merging this PR?

File: src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs (hunk 3 above)
Comment (soninaren) [reply src/WebJobs.Script.WebHost/Middleware/HttpRequestBodySizeMiddleware.cs]: Fixed in the PR

File: src/WebJobs.Script/ScriptConstants.cs (hunk 4 above)
Comment (soninaren) [reply src/WebJobs.Script/ScriptConstants.cs]: It is not required. I have made it long.

**Comment by soninaren** [comment]
> Could you link to the documentation PR before merging this PR?

https://github.com/MicrosoftDocs/azure-docs/issues/72449

**Comment by soninaren** [comment]
@fabiocav , I have addressed all the PR feedback. Can you please take another look?

**Review by fabiocav — APPROVED** [review]

**Pull request merged**